onnx/test/helper_test.py

```diff
@@ -218,6 +218,12 @@ def test_attr_empty_list(self) -> None:
         self.assertEqual(len(attr.strings), 0)
         self.assertRaises(ValueError, helper.make_attribute, "empty", [])
 
+    def test_attr_mismatch(self) -> None:
+        with pytest.raises(TypeError) as te:
```

Comment:
Consider https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex instead.

Reply:
I think the pytest way is clearer, but will change this if its a blocker.

Reply:
Yes. We create tests with unittest and use pytest only as the runner (or in cases when unittest doesn't provide a feature we need)